Source/WebGPU/WebGPU/ShaderModule.mm

```diff
@@ -73,25 +73,35 @@
     return { { *wgsl, hints } };
 }
 
-id<MTLLibrary> ShaderModule::createLibrary(id<MTLDevice> device, const String& msl, String&& label, NSError** error)
+id<MTLLibrary> ShaderModule::createLibrary(id<MTLDevice> device, const String& msl, String&& label, NSError** errorPointer)
 {
     auto options = [MTLCompileOptions new];
 ALLOW_DEPRECATED_DECLARATIONS_BEGIN
     options.fastMathEnabled = YES;
 ALLOW_DEPRECATED_DECLARATIONS_END
-    // FIXME(PERFORMANCE): Run the asynchronous version of this
-    id<MTLLibrary> library = [device newLibraryWithSource:msl options:options error:error];
-    if (error && *error) {
+
+    __block id<MTLLibrary> returnLibrary = nil;
+
+    // Call the asynchronous newLibraryWithSource method
+    [device newLibraryWithSource:msl options:options completionHandler:^(id<MTLLibrary> library, NSError *error) {
```

Comment:
Unless the completion handler runs before the call to `return returnLibrary;` i.e., effectively synchronously, then we would return nil here and the call to `createLibrary` in Pipeline.mm would fail

Reply:
newLibraryWithSource is the function that then creates the function by compiling async it seems. I do not think the completion handler runs async, but rather the actual compilation.

Reply:
I checked the sources, the completion handler runs asynchronously.

Reply:
It should be possible to confirm with a very long or complex shader